chopSEQ.py

```diff
@@ -37,204 +40,232 @@
-	tandem_max_repeat=350
-	tandem_threshold=10
-	tandem_mismatch=5
-	tandem_identity_threshold=85
-	#Used in Step 3:
-	minimum_read_length_threshold=0
-	maximum_read_length_threshold=100000
-	# *********************************/
-
-	try:
-		opts,args=getopt.getopt(argv,"hi:f:r:vl:m:",["input_file","forward_primer","reverse_primer","minimum_length","maximum_length"])
-	except getopt.GetoptError:
-		usage()
-		sys.exit(2)
-	for opt, arg in opts:
-		if opt == '-h':
-			usage()
-			sys.exit()
-		elif opt == '-v':
-			verbosity=1
-		elif opt in ("-i","--input_file"):
-			input_file=arg
-                elif opt in ("-f","--forward_primer"):
-                        forward_primer=arg
-                elif opt in ("-r","--reverse_primer"):
-                        reverse_primer=arg
-		elif opt in ("-l","--minimum_length"):
-			minimum_read_length_threshold=int(arg)
-                elif opt in ("-m","--maximum_length"):
-                        maximum_read_length_threshold=int(arg)
-	if (input_file=='' or forward_primer=='' or reverse_primer==''):
-		usage()
-		sys.exit(2)
 
 	#Some of the references we used in writing the code
 	#Reference: http://biopython.org/DIST/docs/tutorial/Tutorial.html#htoc85
-	#Reference: https://stackoverflow.com/questions/287871/print-in-terminal-with-colors-using-python
+	#Reference: https://stackoverflow.com/questions/287871/print-in-terminal-with-colors-using-pyton
+    #this allow the multiprocessor call and allow for the ability to print out the results
+    with concurrent.futures.ProcessPoolExecutor() as executor:
+        results =[executor.submit(process_seq_records, seq_record, forward_primer, reverse_primer, verbosity, minimum_read_length_threshold, maximum_read_length_threshold) for seq_record in SeqIO.parse(input_file,"fasta")]
 
-	for seq_record in SeqIO.parse(input_file,"fasta"):
-		if verbosity:
-			print('\x1b[6;37;40m' + str(seq_record.id) + '\x1b[0m')
-		forward_orientation_forward_primer_alignment=pairwise2.align.localms(str(seq_record.seq),forward_primer,primer_match_score, primer_mismatch_score, primer_open_gap_score, primer_extend_gap_score, one_alignment_only=1)
-		forward_orientation_reverse_primer_alignment=pairwise2.align.localms(str(seq_record.seq),str(Seq(reverse_primer).reverse_complement()),primer_match_score,primer_mismatch_score, primer_open_gap_score, primer_extend_gap_score, one_alignment_only=1)
-                reverse_orientation_forward_primer_alignment=pairwise2.align.localms(str(seq_record.seq),reverse_primer,primer_match_score,primer_mismatch_score, primer_open_gap_score, primer_extend_gap_score, one_alignment_only=1)
-                reverse_orientation_reverse_primer_alignment=pairwise2.align.localms(str(seq_record.seq),str(Seq(forward_primer).reverse_complement()),primer_match_score,primer_mismatch_score, primer_open_gap_score, primer_extend_gap_score, one_alignment_only=1)
+        for resultant in concurrent.futures.as_completed(results):
+            print(resultant.result())
```

Comment:
This is microprocessing changing list comprehension for the reading in of the file in.